FreeRTOS/Test/CMock/smp/Makefile

```diff
@@ -8,6 +8,7 @@ include ../makefile.in
 SUITES	+=	single_priority_no_timeslice
 SUITES	+=	single_priority_timeslice
 SUITES	+=	multiple_priorities_no_timeslice
+SUITES	+=	multiple_priorities_no_timeslice_tickless_idle
```

Comment:
The following test should be in the tickless_idle dir: test_coverage_xTaskGenericNotify_with_eAction_equalto_eSetValueWithoutOverwrite_branch_taskWAITING_NOTIFICATION

Reply:
Sorry. I forget the other file covg_multiple_priorities_no_timeslice.c. However, the test cases "test_coverage_xTaskGenericNotify_with_eAction_equalto_eSetValueWithoutOverwrite_branch_taskWAITING_NOTIFICATION" doesn't need the macro "configUSE_TICKLESS_IDLE = 1". Suggest we remove move the test case to multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice.c.

Reply:
This subset of the function xTaskGenericNotify is covered by an #ifdef configUSE_TICKLESS_IDLE:

```
      /* If the task is in the blocked state specifically to wait for a
             * notification then unblock it now. */
            if( ucOriginalNotifyState == taskWAITING_NOTIFICATION )
            {
                listREMOVE_ITEM( &( pxTCB->xStateListItem ) );
                prvAddTaskToReadyList( pxTCB );

                /* The task should not have been on an event list. */
                configASSERT( listLIST_ITEM_CONTAINER( &( pxTCB->xEventListItem ) ) == NULL );

                #if ( configUSE_TICKLESS_IDLE != 0 )
                {
                    /* If a task is blocked waiting for a notification then
                     * xNextTaskUnblockTime might be set to the blocked task's time
                     * out time.  If the task is unblocked for a reason other than
                     * a timeout xNextTaskUnblockTime is normally left unchanged,
                     * because it will automatically get reset to a new value when
                     * the tick count equals xNextTaskUnblockTime.  However if
                     * tickless idling is used it might be more important to enter
                     * sleep mode at the earliest possible time - so reset
                     * xNextTaskUnblockTime here to ensure it is updated at the
                     * earliest possible time. */
                    prvResetNextTaskUnblockTime();
                }
                #endif
```

Reply:
The code section is already covered by other test cases.
```
                #if ( configUSE_TICKLESS_IDLE != 0 )
                {
                    /* If a task is blocked waiting for a notification then
                     * xNextTaskUnblockTime might be set to the blocked task's time
                     * out time.  If the task is unblocked for a reason other than
                     * a timeout xNextTaskUnblockTime is normally left unchanged,
                     * because it will automatically get reset to a new value when
                     * the tick count equals xNextTaskUnblockTime.  However if
                     * tickless idling is used it might be more important to enter
                     * sleep mode at the earliest possible time - so reset
                     * xNextTaskUnblockTime here to ensure it is updated at the
                     * earliest possible time. */
                    prvResetNextTaskUnblockTime();
                }
                #endif
```
Can you describe more why are we adding the test case in covg_multiple_priorities_no_timeslice_tickless_idle.c file?

Reply:
It may produce different side-effects due to the call to prvResetNextTaskUnblockTime(). I am not yet asserting on the differences post-test but could.